projects/hiptensor/library/src/contraction/contraction_solution_impl.hpp

```diff
@@ -148,15 +150,15 @@ namespace hiptensor
                 return std::vector<ck::index_t>(v.begin(), v.end());
             };
 
-            using hiptensor::HiptensorOptions;
-            auto& options = HiptensorOptions::instance();
-            if(options->isColMajorStrides()) 
-            {
+            //Apply CK ColMajor strides for col major layout
+            if(isColMajorLayout(a_ms_ks_strides, a_ms_ks_lengths))
                 normal_a_ms_ks_strides = getCKColMajorStrides(normal_a_ms_ks_lengths);
+            if(isColMajorLayout(b_ns_ks_strides, b_ns_ks_lengths))
                 normal_b_ns_ks_strides = getCKColMajorStrides(normal_b_ns_ks_lengths);
+            if(isColMajorLayout(ds_ms_ns_strides, ds_ms_ns_lengths))
                 normal_ds_ms_ns_strides = getCKColMajorStrides(normal_ds_ms_ns_lengths);
+            if(isColMajorLayout(e_ms_ns_strides, e_ms_ns_lengths))
                 normal_e_ms_ns_strides = getCKColMajorStrides(normal_e_ms_ns_lengths);
```

Comment:
The stride optimization is only valid if applied to all Tensors in the contraction.
```suggestion
            if(isColMajorLayout(a_ms_ks_strides, a_ms_ks_lengths) && isColMajorLayout(b_ns_ks_strides, b_ns_ks_lengths) && isColMajorLayout(ds_ms_ns_strides, ds_ms_ns_lengths) && isColMajorLayout(e_ms_ns_strides, e_ms_ns_lengths))
            {
                normal_a_ms_ks_strides = getCKColMajorStrides(normal_a_ms_ks_lengths);
                normal_b_ns_ks_strides = getCKColMajorStrides(normal_b_ns_ks_lengths);
                normal_ds_ms_ns_strides = getCKColMajorStrides(normal_ds_ms_ns_lengths);
                normal_e_ms_ns_strides = getCKColMajorStrides(normal_e_ms_ns_lengths);
            }
```

Reply:
Thanks for the suggestion. However, it doesn’t require all tensors to be column-major. I verified this both in the contraction benchmark and with individual examples. Even when only some of the tensors are column-major, there’s still a substantial performance boost. For instance, if only tensor A is column-major and the others (B, C, etc.) are row-major—or vice versa—adjusting the strides can still yield a 2–10× speedup.

Reply:
But is it valid? Does it generate valid results? We don't have tests that perform data validation with mixed memory layouts. I don't think it's safe to perform this operation in such case unless we have tests to confirm it actually works.

According to the analysis I had made for the memory layout modifications we are doing, the contraction only remained valid because this specific modification was being applied to all of the tensors.

Reply:
Yes, it’s valid. I tested the benchmark using mixed layouts (by modifying the code) and verified all results against the CPU. Every test in the mixed-layout configuration was confirmed to match the CPU outputs.

<img width="1901" height="903" alt="image" src="https://github.com/user-attachments/assets/6e4c8ddc-030d-496c-a72c-f2ab5edfc037" />

Reply:
Nice!